src/js/components/modals/ServiceFormModal.js

```diff
@@ -269,6 +269,35 @@ class ServiceFormModal extends mixin(StoreMixin) {
   shouldDisableForm(service) {
     let containerSettings = service.getContainerSettings();
 
+    let portDefinitions = service.getPortDefinitions();
+    let name = new RegExp(`${service.getId()}`, 'g');
+
+    if (portDefinitions) {
+      if (!portDefinitions.some(function (port) {
+        if (port.labels == null || Object.keys(port.labels).length === 0) {
+          return true;
+        }
+        return Object.keys(port.labels).some(function (key) {
+          return name.test(port.labels[key]);
+        });
+      })) {
+        return true;
+      }
+    }
+
+    if (containerSettings && containerSettings.docker && containerSettings.docker.portMappings) {
+      if (!containerSettings.docker.portMappings.some(function (port) {
+        if (port.labels == null || Object.keys(port.labels).length === 0) {
+          return true;
+        }
+        return Object.keys(port.labels).some(function (key) {
+          return name.test(port.labels[key]);
```

Comment:
![](https://cloud.githubusercontent.com/assets/647035/18007923/4cc8f67c-6ba6-11e6-97b3-50409811ef2c.gif)
Let's test for the complete string, as this would otherwise allow partial matches which is not correct.